src/packet/public_key_encrypted_session_key.js

```diff
@@ -67,13 +72,29 @@ class PublicKeyEncryptedSessionKeyPacket {
    * @param {Uint8Array} bytes - Payload of a tag 1 packet
    */
   read(bytes) {
-    this.version = bytes[0];
-    if (this.version !== VERSION) {
+    let offset = 0;
+    this.version = bytes[offset++];
+    if (this.version !== 3 && this.version !== 6) {
       throw new UnsupportedError(`Version ${this.version} of the PKESK packet is unsupported.`);
     }
-    this.publicKeyID.read(bytes.subarray(1, bytes.length));
-    this.publicKeyAlgorithm = bytes[9];
-    this.encrypted = crypto.parseEncSessionKeyParams(this.publicKeyAlgorithm, bytes.subarray(10));
+    if (this.version === 6) {
+      this.publicKeyVersion = bytes[offset++];
+      const fingerprintLength = this.publicKeyVersion ? (this.publicKeyVersion >= 5 ? 32 : 20) : 0;
+      this.publicKeyFingerprint = bytes.subarray(offset, offset + fingerprintLength);
+      offset += fingerprintLength;
+      if (this.publicKeyVersion === 0) {
+        this.publicKeyID = KeyID.wildcard();
+      } else if (this.publicKeyVersion >= 5) {
+        this.publicKeyID.read(this.publicKeyFingerprint);
```

Comment:
Here it should be `this.publicKeyID.read(this.publicKeyFingerprint.subarray(-8));`
For v6 keys the Key ID is the high-order 64 bits of the fingerprint.

Reply:
NVM is correct.